src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -8,14 +9,19 @@ import com.typesafe.config.ConfigFactory
 import net.ceedubs.ficus.Ficus._
 import org.broadinstitute.dsde.workbench.leonardo.config.DataprocConfig
 import org.broadinstitute.dsde.workbench.leonardo.model.ClusterStatus.ClusterStatus
-import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.{GoogleBucket, GoogleProject, GoogleServiceAccount}
+import org.broadinstitute.dsde.workbench.leonardo.model.TypedString.LabelMap
 import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat}
 
-// maybe we want to get fancy later
-object ModelTypes {
-  type GoogleProject = String
-  type GoogleServiceAccount = String
-  type GoogleBucket = String
+sealed trait TypedString extends Any
+case class ClusterName(s: String) extends AnyVal with TypedString
+case class GoogleProject(s: String) extends AnyVal with TypedString
+case class GoogleServiceAccount(s: String) extends AnyVal with TypedString
+case class GoogleBucket(s: String) extends AnyVal with TypedString
+case class OperationName(s: String) extends AnyVal with TypedString
+case class IP(s: String) extends AnyVal with TypedString
```

Comment:
Not sure I like `s` because I confuse it with string interpolation `s"Hello $name"`. Maybe `.string`?